src/signTxMint.c

```diff
@@ -24,7 +24,7 @@ static inline void advanceState()
 	switch (subctx->state) {
 
 	case STATE_MINT_TOP_LEVEL_DATA:
-		ASSERT(subctx->numAssetGroups > 0);
+		VALIDATE(subctx->numAssetGroups > 0, ERR_INVALID_DATA);
```

Comment:
The assertion should stay here, it only ensures that the assumptions under which the state machine works are satisfied. If it was zero at this point, we have a bug somewhere.

Reply:
True, we should validate at the point of reading it.